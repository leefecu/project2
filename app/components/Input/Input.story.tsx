import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import Input from './Input';
import Styles from '../../Stories.styles';

storiesOf('Component | Input', module).add('Primary', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>Default</Text>
        <Input
          clearText={action('clear text')}
          number=""
          onUpdate={action('update number')}
          placeholder="Enter mobile number"
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>with number</Text>
        <Input
          clearText={action('clear text')}
          number="+62 22 1234 5678"
          onUpdate={action('update number')}
          placeholder="Enter mobile number"
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>with error</Text>
        <Input
          clearText={action('clear text')}
          invalid={true}
          number="09 373 5"
          onUpdate={action('update number')}
          placeholder="Enter mobile number"
        />
      </View>
    </View>
  );
});
