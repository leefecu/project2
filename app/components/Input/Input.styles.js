import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.white,
    borderColor: Colors.white,
    borderRadius: Metrics.inputBorderRadius,
    borderWidth: 1,
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: Metrics.basePadding,
    paddingHorizontal: Metrics.basePadding,
    width: Metrics.width * 0.9,

    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  errorIcon: {
    flex: 1,
  },
  clearTextIcon: {
    alignItems: 'flex-end',
    flex: 1,
  },
  hasError: {
    borderColor: Colors.errorBorder,
    borderWidth: 1,
  },
  input: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    flex: 8,
    padding: 0,
    margin: 0,
  },
});
