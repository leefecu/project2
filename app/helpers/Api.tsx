import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
