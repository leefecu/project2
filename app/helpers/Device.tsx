import { NativeModules, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import CarrierInfo from 'react-native-carrier-info';
import RNFS from 'react-native-fs';

const NZ_MCC = '530';
const NZ_COUNTRY_CODE = 'nz';
const IS_IOS = Platform.OS === 'ios';

const NetworkInfo = NativeModules.RNNetworkInfo;

export const isSimulator = async (): Promise<boolean> => {
  const isEmulator = await DeviceInfo.isEmulator();
  return isEmulator;
};

export const isSimIn = async () => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return true;
  }

  try {
    // mobileCountryCode method fails when no SIM card in the device
    await CarrierInfo.mobileCountryCode();
    return true;
  } catch (e) {
    return false;
  }
};

export const checkForRoaming = async (debug: boolean = false): Promise<boolean> => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return debug;
  }

  try {
    if (Platform.OS === 'ios') {
      const operatorPlist = await NativeModules.NetworkInfo.getOperatorPlist();
      const mcc = operatorPlist.match(/\d+\/carrier.plist/g);
      if (mcc && mcc.length > 0) {
        return mcc[0].substr(0, 3) !== NZ_MCC;
      }
      return false;
    }
    const countryCode = await NetworkInfo.isoCountryCode();
    return countryCode !== NZ_COUNTRY_CODE;
  } catch (e) {
    return false;
  }
};

const majorVersionIOS: number = parseInt(`${Platform.Version}`, 10);

export const isAutoFillSupported: boolean = IS_IOS && majorVersionIOS >= 12;

// Figured out how current(old) DataClock app stores msisdn
// it's stored in xml file
// shared_prefs/nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml
const msisdnFileName = 'nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml';
const msisdnFolderName = 'shared_prefs';
const msisdnRegex = /com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">\d+<\/string/g;

export const getExistingMsisdn = async (): Promise<TMigrationData> => {
  if (IS_IOS) {
    try {
      const msisdn = await NativeModules.GetNumber.getMsisdn();
      return { number: msisdn, upgrade: !!msisdn };
    } catch (e) {
      return { number: '', upgrade: false };
    }
  }
  // This method only works with Android
  try {
    // e.g. RNFS.DocumentDirectoryPath = '/data/user/0/nz.co.twodegreesmobile.comptelfwd.smpl/files',
    const paths = RNFS.DocumentDirectoryPath.split('/');
    // need to remove "/files" and replace with "shared_prefs"
    paths[paths.length - 1] = msisdnFolderName;
    const path = paths.join('/');
    const xmlPath = `${path}/${msisdnFileName}`;
    const existingAccount = await RNFS.readFile(xmlPath);
    const match = msisdnRegex.exec(existingAccount);
    if (match && match.length > 0) {
      const msisdn = match[0]
        .replace('com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">', '')
        .replace('</string', '');

      return { number: msisdn, upgrade: true };
    }
    return { number: '', upgrade: false };
  } catch (e) {
    return { number: '', upgrade: false };
  }
};
