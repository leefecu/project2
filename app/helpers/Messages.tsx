import { TERROR_MESSAGE, ERROR_CODE, ERROR_MESSAGES } from '../constants/Messages';
import { ACTION_TYPE } from '../constants/Shared';

export const getErrorMessage = (errorCode: ERROR_CODE, action?: ACTION_TYPE): TERROR_MESSAGE => {
  switch (errorCode) {
    case ERROR_CODE.ACCOUNT_HOLIDAY:
    case ERROR_CODE.INVALID_ACCOUNT_STATUS: {
      if (action === ACTION_TYPE.REGISTER) {
        return ERROR_MESSAGES.NOT_ELIGIBLE;
      }
      return ERROR_MESSAGES.ACCOUNT_SUSPENDED;
    }
    case ERROR_CODE.CREDIT_BARRED:
      return ERROR_MESSAGES.CREDIT_LIMIT;
    case ERROR_CODE.EXPIRED:
      return ERROR_MESSAGES.EXPIRED;
    case ERROR_CODE.INSUFFICIENT_CREDIT:
      return ERROR_MESSAGES.LOW_BALANCE;
    case ERROR_CODE.INVALID_PASSCODE:
      return ERROR_MESSAGES.INVALID_PASSCODE;
    case ERROR_CODE.INVALID_ACCOUNT:
    case ERROR_CODE.INVALID_ACCOUNT_TYPE:
    case ERROR_CODE.INVALID_ROLE:
      return ERROR_MESSAGES.ROLE_PERMISSION;
    case ERROR_CODE.BLOCKED: {
      if (action === ACTION_TYPE.REGISTER) {
        return ERROR_MESSAGES.NOT_ELIGIBLE;
      }
      if (action === ACTION_TYPE.PURCHASE) {
        return ERROR_MESSAGES.ACCOUNT_SUSPENDED;
      }
      return ERROR_MESSAGES.NOT_ELIGIBLE;
    }
    case ERROR_CODE.INVALID_PLAN:
      return ERROR_MESSAGES.NOT_ELIGIBLE;
    case ERROR_CODE.DEVICE_NOT_MATCH:
      return ERROR_MESSAGES.DEVICE_NOT_MATCH;
    case ERROR_CODE.NOT_REGISTERED:
      return ERROR_MESSAGES.NOT_REGISTERED;
    case ERROR_CODE.SPEND_LIMIT_REACHED:
      return ERROR_MESSAGES.SPEND_LIMIT;
    case ERROR_CODE.TOO_MANY_ATTEMPT:
      return ERROR_MESSAGES.TOO_MANY_ATTEMPT;
    case ERROR_CODE.PACK_STACK_LIMIT_REACHED:
      return ERROR_MESSAGES.PACK_STACK_LIMIT_REACHED;
    case ERROR_CODE.QD_ISSUE: {
      if (action === ACTION_TYPE.DASHBOARD) {
        return ERROR_MESSAGES.NOT_ELIGIBLE;
      }
      return ERROR_MESSAGES.GENERIC;
    }
    default:
      return ERROR_MESSAGES.GENERIC;
  }
};
