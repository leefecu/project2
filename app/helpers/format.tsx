export const getFormattedMobileNumber = (number: string): string => {
  const prefix = '+64';
  const numberParts = [];
  if (number.startsWith('+64')) {
    numberParts.push(number.substring(3, 5));
    numberParts.push(number.substring(5, 8));
    numberParts.push(number.substring(8, 13));
  } else if (number.startsWith('64')) {
    numberParts[0] = number.substring(2, 4);
    numberParts.push(number.substring(4, 7));
    numberParts.push(number.substring(7, 12));
  } else if (number.startsWith('0064')) {
    numberParts[0] = number.substring(4, 6);
    numberParts.push(number.substring(6, 9));
    numberParts.push(number.substring(9, 14));
  } else if (number.startsWith('02')) {
    numberParts[0] = number.substring(1, 3);
    numberParts.push(number.substring(3, 6));
    numberParts.push(number.substring(6, 12));
  } else if (number.startsWith('2')) {
    numberParts[0] = number.substring(0, 2);
    numberParts.push(number.substring(2, 5));
    numberParts.push(number.substring(5, 12));
  }

  return `${prefix} ${numberParts.join(' ')}`;
};

// number is always in the format of 22xxxyyyy
export const getLocalFormattedMobileNumber = (number: string): string => {
  return `0${number.substring(0, 2)} ${number.substring(2, 13)}`;
};

export const parseDuration = (duration: number): number =>
  duration % 60 === 0 ? duration / 60 : duration;

export const getUnit = (duration: number, short: boolean = false): string => {
  if (duration % 60 === 0) {
    const hour = duration / 60;
    return `${short ? 'hr' : 'hour'}${hour > 1 ? 's' : ''}`;
  }
  return `${short ? 'min' : 'minute'}${duration > 1 ? 's' : ''}`;
};

const DAY = 86400;
const HOUR = 3600;
const MIN = 60;

export const gerFormatedRemainingTime = (time: number): string => {
  let remainingTime = Math.ceil(Number(time));
  let displayFormatFactor = 0;
  const timeTokens = [];
  if (remainingTime >= DAY) {
    timeTokens.push(`${Math.floor(remainingTime / DAY)}d`);
    remainingTime %= DAY;
    displayFormatFactor = DAY;
  }
  if (remainingTime >= HOUR) {
    timeTokens.push(`${Math.floor(remainingTime / HOUR)}h`);
    remainingTime %= HOUR;
    if (!displayFormatFactor) {
      displayFormatFactor = HOUR;
    }
  }
  if (displayFormatFactor <= HOUR && timeTokens.length < 2 && remainingTime >= MIN) {
    timeTokens.push(`${Math.floor(remainingTime / MIN)}m`);
    remainingTime %= MIN;
    if (!displayFormatFactor) {
      displayFormatFactor = MIN;
    }
  }
  if (displayFormatFactor <= MIN && timeTokens.length < 2 && remainingTime > 0) {
    timeTokens.push(`${remainingTime}s`);
  }
  return timeTokens.join(' ');
};
