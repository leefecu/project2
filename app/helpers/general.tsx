import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
