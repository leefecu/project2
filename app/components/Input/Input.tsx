import React, { createRef, PureComponent } from 'react';
import { TextInput, TouchableHighlight, View } from 'react-native';

import { noop } from '../../helpers/general';

import Styles from './Input.styles';
import Colors from '../../themes/Colors';

import ClearTextSVG from '../../images/clearText.svg';
import ErrorSVG from '../../images/error.svg';

interface DefaultProps {
  autoFocus?: boolean;
  invalid?: boolean;
  number: string;
  onBlur?: () => void;
  onUpdate: (number: string) => void;
  placeholder: string;
  returnKeyType:
    | 'go'
    | 'none'
    | 'search'
    | 'default'
    | 'done'
    | 'next'
    | 'send'
    | 'previous'
    | 'google'
    | 'join'
    | 'route'
    | 'yahoo'
    | 'emergency-call'
    | undefined;
}

interface Props extends DefaultProps {
  clearText: () => void;
}

class Input extends PureComponent<Props> {
  textInput = createRef<TextInput>();

  static defaultProps: DefaultProps = {
    autoFocus: false,
    invalid: false,
    number: '',
    onBlur: noop,
    onUpdate: noop,
    placeholder: 'placeholder',
    returnKeyType: 'go',
  };

  render() {
    const {
      autoFocus,
      clearText,
      invalid,
      number,
      onBlur,
      onUpdate,
      placeholder,
      ...props
    } = this.props;
    return (
      <View style={Styles.container}>
        {invalid && (
          <View style={Styles.errorIcon}>
            <ErrorSVG />
          </View>
        )}
        <TextInput
          ref={this.textInput}
          autoFocus={autoFocus}
          allowFontScaling={true}
          keyboardType="phone-pad"
          onBlur={onBlur}
          onChangeText={onUpdate}
          placeholder={placeholder}
          placeholderTextColor={Colors.placeholder}
          style={Styles.input}
          value={number}
          {...props}
        />
        {number.length > 0 && (
          <TouchableHighlight
            activeOpacity={0.5}
            onPress={clearText}
            style={Styles.clearTextIcon}
            testID="clearTextButton"
            underlayColor={Colors.transparent}>
            <ClearTextSVG />
          </TouchableHighlight>
        )}
      </View>
    );
  }
}

export default Input;
