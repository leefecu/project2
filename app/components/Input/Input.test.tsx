/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Input from './Input';

import { noop } from '../../helpers/general';

describe('Input component', () => {
  test('render default input', () => {
    const button = create(
      <Input clearText={noop} number="" onUpdate={noop} placeholder="Enter mobile number" />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render input with number', () => {
    const button = create(
      <Input
        clearText={noop}
        number="+62 22 1234 5678"
        onUpdate={noop}
        placeholder="Enter mobile number"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render input with invalid number & error', () => {
    const button = create(
      <Input
        clearText={noop}
        invalid={true}
        number="09 373 5"
        onUpdate={noop}
        placeholder="Enter mobile number"
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('trigger clearText function upon press clear text icon', () => {
    const clearTextEvent = jest.fn();
    const component = shallow(
      <Input
        clearText={clearTextEvent}
        number="+62 22 1234 5678"
        onUpdate={noop}
        placeholder="Enter mobile number"
      />,
    );
    component
      .find('TouchableHighlight')
      .first()
      .simulate('press');
    expect(clearTextEvent).toHaveBeenCalled();
  });
});
