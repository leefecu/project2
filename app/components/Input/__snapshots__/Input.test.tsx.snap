// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render default input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 5,
      "flexDirection": "row",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 675,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={false}
    keyboardType="phone-pad"
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter mobile number"
    placeholderTextColor="#b3b3b3"
    rejectResponderTermination={true}
    returnKeyType="go"
    style={
      Object {
        "color": "#004071",
        "flex": 8,
        "fontSize": 17,
        "margin": 0,
        "padding": 0,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`Input component render input with invalid number & error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 5,
      "flexDirection": "row",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 675,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SVGMock />
  </View>
  <TextInput
    allowFontScaling={true}
    autoFocus={false}
    keyboardType="phone-pad"
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter mobile number"
    placeholderTextColor="#b3b3b3"
    rejectResponderTermination={true}
    returnKeyType="go"
    style={
      Object {
        "color": "#004071",
        "flex": 8,
        "fontSize": 17,
        "margin": 0,
        "padding": 0,
      }
    }
    underlineColorAndroid="transparent"
    value="09 373 5"
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
    testID="clearTextButton"
  >
    <SVGMock
      style={null}
    />
  </View>
</View>
`;

exports[`Input component render input with number 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 5,
      "borderWidth": 1,
      "elevation": 5,
      "flexDirection": "row",
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 675,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={false}
    keyboardType="phone-pad"
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter mobile number"
    placeholderTextColor="#b3b3b3"
    rejectResponderTermination={true}
    returnKeyType="go"
    style={
      Object {
        "color": "#004071",
        "flex": 8,
        "fontSize": 17,
        "margin": 0,
        "padding": 0,
      }
    }
    underlineColorAndroid="transparent"
    value="+62 22 1234 5678"
  />
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
    testID="clearTextButton"
  >
    <SVGMock
      style={null}
    />
  </View>
</View>
`;
